src/managers/RoleManager.js

```diff
@@ -244,6 +262,21 @@ class RoleManager extends CachedManager {
     }).guild;
   }
 
+  /**
+   * Compares the positions of two roles.
+   * @param {RoleResolvable} role1 First role to compare
+   * @param {RoleResolvable} role2 Second role to compare
+   * @returns {number} Negative number if the first role's position is lower (second role's is higher),
+   * positive number if the first's is higher (second's is lower), 0 if equal
+   */
+  comparePositions(role1, role2) {
+    const resolvedRole1 = this.resolve(role1);
+    const resolvedRole2 = this.resolve(role2);
+    if (!resolvedRole1 || !resolvedRole2) throw new TypeError('INVALID_TYPE', 'role', 'Role nor a Snowflake');
+    if (resolvedRole1.position === resolvedRole2.position) return resolvedRole2.id - resolvedRole1.id;
```

Comment:
Should we maybe BigInt the ids when doing a `-`? CC @kyranet

Reply:
I don't think it matters here, the strings are casted to numbers (`'4' - '3'` returns `1`) before the operation. Since only the 2 digits will be imprecise, and they're always defined by the last field (increment, 12 bits) by a small set of them, it'd require the same worker and process to create two roles in the exact same millisecond and within a few hundred ops, which seem extremely unlikely.

However, if you want correctness and handling that extreme edge-case, be my guest

Reply:
Handled nonetheless!